### ISSUE TITLE
Problems with square source images in JIT 2

### ISSUE THREAD

**Issue by animaux** [symphonycms/jit_image_manipulation#120]
Apparently JIT 2 chokes on square images in Mode 4 if the output is square too:

If the source image is 800×800px:

  …/image/4/400/400/img/bilder/schelling-von-halbig-1-1459074525.jpg

this render the invalid dimensions error, whereas

…/image/4/400/401/img/bilder/schelling-von-halbig-1-1459074525.jpg

works as expected.

![hueylhtbs5510406786331740](https://cloud.githubusercontent.com/assets/446874/14877467/896cd298-0d1d-11e6-9ec6-a4223c024c94.jpg)

**Comment by animaux** [comment]
When using the same settings in a recipe Symphony throws:

    Symphony Fatal Error: No JIT filter was found for this request.

Not sure if this is a result of the error above or a different issue.

Reverting back to `JIT 1.4.4` (including the .htaccess-rules) solves both issues. Which also leads me to the point that the rewrite rules in the readme should be updated to the current version in 2.0.0!

**Comment by nitriques** [comment]
> Which also leads me to the point that the rewrite rules in the readme should be updated to the current version in 2.0.0!

Done in c7cd618

> this render the invalid dimensions error, whereas

Done in 3d92d82

Would you mind testing against the `integration` branch to see if the "no filter found" error persists ?

Thanks!

**Comment by nitriques** [comment]
@brendo Would you mind reviewing 3d92d82 ? Thanks!

**Comment by animaux** [comment]
Sure, will try tomorrow!

**Comment by nitriques** [comment]
Thanks @animaux

**Comment by animaux** [comment]
Sorry to report: All above mentioned problems persist in `integration`.

**Comment by animaux** [comment]
Plus in `mode 2` JIT seems to ignore the smaller value. Weird.

**Comment by nitriques** [comment]
Well I do not know what to do as I can't seem to reproduce the error anymore. I'll check the mode 2 ASAP.

**Comment by nitriques** [comment]
@animaux Can you please test the latest `integration` branch please ? Thanks!

**Comment by animaux** [comment]
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (image.php, line 0)

I now get links like:

    http://…/extensions/jit_image_manipulation/lib/image.php/?param=2/136/136/5/img/bilder/20080718_25-b-pikto-800-neu-1420714412.jpg

Following these links results in a 500.

**Comment by nitriques** [comment]
Yeah, can you please update your .htaccess rule please ? Is this a new install ?

```
RewriteRule ^image\/(.+)$ index.php?mode=jit&param=$1 [B,L,NC]
```

**Comment by animaux** [comment]
URLs now look better, I still get 500s for each image though. Will check the logs.

**Comment by nitriques** [comment]
Hum... Anything in php's `error_log`?

**Comment by animaux** [comment]
It’s running `PHP 5.6.10`

From the Symphony-Log:

    07. June 2016 16:43 > UNKNOWN: ErrorException 0 - Illegal string offset 'allow_origin' on line 289 of /…/extensions/jit_image_manipulation/lib/class.jit.php

**Comment by nitriques** [comment]
Ah damn it. I'll fix this ASAP. Been testing it mostly on old 5.3. Bad me.

**Comment by animaux** [comment]
No worries. Thanks!

**Comment by nitriques** [comment]
@animaux can you test ae4de7 ? Thanks!

**Comment by animaux** [comment]
I’ve tried Mode 2. Looking good so far.

**Comment by animaux** [comment]
From `integration`

**Comment by nitriques** [comment]
Great!
